### PR TITLE
fix middlewares

### PR DESCRIPTION
**Pull request by Digital39999**
Type: Bugfix
Fixes middleware returns.

### CODE REVIEW

File: packages/eventsub-http/src/EventSubHttpBase.ts [hunk 1]
```diff
@@ -211,16 +211,17 @@ export abstract class EventSubHttpBase extends EventSubBase {
...
 					this._logger.debug(`Old notification(s) prevented for event: ${id}`);
-				} else {
-					const payload = data as EventSubNotificationPayload;
-					if ('events' in payload) {
-						for (const event of payload.events) {
-							this._handleSingleEventPayload(subscription, event.data, event.id);
-						}
-					} else {
-						this._handleSingleEventPayload(subscription, payload.event, messageId);
+				}
```
Comment (d-fischer) [inline packages/eventsub-http/src/EventSubHttpBase.ts:214]: This `else` was *not* unnecessary. Removing it renders the above check useless.

File: packages/eventsub-http/src/EventSubHttpBase.ts [hunk 2]
```diff
@@ -258,8 +258,6 @@ export abstract class EventSubHttpBase extends EventSubBase {
 				res.setHeader('Content-Type', 'text/plain');
 				res.writeHead(410);
 				res.end('Not OK');
-			} else {
-				next();
```
Comment (d-fischer) [inline packages/eventsub-http/src/EventSubHttpBase.ts:262]: Why remove this? You're leaving the request hanging.

**Review by d-fischer — CHANGES_REQUESTED** [review]

File: packages/eventsub-http/src/EventSubHttpBase.ts (hunk 1 above)
Comment (Digital39999) [reply packages/eventsub-http/src/EventSubHttpBase.ts:214]: Oops, you're right, reverting.

**Comment by d-fischer** [comment]
You have again based this PR on the `main` branch. If you, as you said, need this to be released quick, base it on `release/7.0` instead, as the PR template even says.

File: packages/eventsub-http/src/EventSubHttpBase.ts (hunk 2 above)
Comment (Digital39999) [reply packages/eventsub-http/src/EventSubHttpBase.ts:262]: Shouldn't that ultimately be handled by user with custom 404 hooks?

Comment (d-fischer) [reply packages/eventsub-http/src/EventSubHttpBase.ts:262]: I don't think these will run if you don't call `next`. The request isn't finished here, so the middleware chain needs to continue.

**Comment by Digital39999** [comment]
What's the actual difference? I see released being made from main, and versions/7 being outdated behind main.

**Comment by d-fischer** [comment]
The `main` branch is where the active development of the next minor (or major) version happens. Until the final release of version 7.1, only prerelease versions will be released from there, as I did today.

7.0.x patches are going on the `versions/7.0` branch.

**Comment by Digital39999** [comment]
Does it really make a significant difference, all other pr's were based on main, and this should be released quickly imo, as it causes issues in production environments.

**Comment by d-fischer** [comment]
If you want it to be released quickly, again, as I said, `main` is the wrong place. If you want this to be merged into `main`, it will be released whenever I finish the stable release of version 7.1, which may take weeks or even months, unless you like to use unstable prerelease versions. `versions/7.0` will get you a stable release some time this week.

**Review by d-fischer — CHANGES_REQUESTED** [review]
Please make sure that your code *actually* bases on the `versions/7.0` branch. Currently, you're pulling the `main` branch changes into the version branch.

**Comment by Digital39999** [comment]
It is?

**Comment by d-fischer** [comment]
It does not, otherwise all the package.json changes wouldn't be in the changes tab.

**Comment by Digital39999** [comment]
There's no way now to change the branch.

**Comment by d-fischer** [comment]
You can always rebase using `git rebase`. I will not accept this PR in this state, as it completely messes up the versioning system I'm using.

**Comment by Digital39999** [comment]
Then just fix this next thing yourself, its annoying already, always errors on startup or any action.

**Pull request closed without merging**